docs/porting/target/tickless/tickless.md

```diff
@@ -0,0 +1,52 @@
+# Tickless mode
+
+Tickless mode is an optimization mechanism available in RTOS for suspending the SysTick. You can use it in situations when RTOS is idle for multiple ticks, so you can save power by entering uninterrupted sleep. 
+
+## Scheduling and sleep modes in Mbed OS
+
+Mbed OS uses the SysTick timer in periods of 1ms to process threads' scheduling.
+
+For instance, a system running two threads would see this timing:
+
+![](resources/Normal_Tick.png)
+
+Note that the device never enters deep sleep and wastes cycles in SysTick while all threads are asleep.
+
+## Tickless mode
+
+To support tickless mode in Mbed OS, your target needs to meet two requirements:
+
+- Support for Sleep HAL API.
+- Support for either Low Power or Microsecond Ticker HAL API.
+
+To enable tickless mode for your target, add the `MBED_TICKLESS` macro in `target.json` (in your target's section):
+
+```json
+"macros_add": [
+    "MBED_TICKLESS"
+]
+```
+
+When tickless mode is enabled, the device's default SysTick mechanism overrides by the [OsTimer](../mbed-os-api-doxy/structos__timer__def.html) implementation, based on either the [low power ticker](../mbed-os-api-doxy/group__hal__lp__ticker.html) or [microsecond ticker](../mbed-os-api-doxy/group__hal__us__ticker.html).
```

Comment:
"overrides by the" is wrong - did you mean "overrides the OsTimer" or "overridden by the OsTimer"?

Reply:
"is overridden by the OsTimer". But "replaced" would be better, I think.

And it's not the "device's default mechanism" - it's Mbed OS's default mechanism (if you don't define MBED_TICKLESS).

The device's default mechanism will be determined by whether or not its target.json defines MBED_TICKLESS. An application can override that in their own mbed_app.json.

Reply:
```suggestion
When tickless mode is enabled, Mbed OS's default [OsTimer](../mbed-os-api-doxy/structos__timer__def.html) replaces SysTick, based on either the [low power ticker](../mbed-os-api-doxy/group__hal__lp__ticker.html) or [microsecond ticker](../mbed-os-api-doxy/group__hal__us__ticker.html).
```

Reply:
I think this needs to be broken. The bit about the lower power/microsecond ticker should be in the next paragraph, where we explain the default and how to override it.

Reply:
As for the device default - should we clarify line 22?